### PR TITLE
Deploy whisper with vllm

### PR DESCRIPTION
**Pull request by nidhihrmth**
Training changes:
1. Clean config.py to include checkpointing, node_count
2. Change tokenizer in seq2seq trainer because tokenizer.processor doesn't save preprocessor_config.json - similar to issue [here](https://discuss.huggingface.co/t/trouble-loading-hf-communitys-openai-whisper-models/78274/3)

Deployment:
1. Example deployment with vllm

Proof:
https://huggingface.co/baseten-admin/whisper-larger-v3-turbo-minirun3/tree/main contains preprocessor_config.json

Deployment working:
image

### CODE REVIEW

File: examples/whisper/inference/config.yaml [hunk 1]
```diff
@@ -0,0 +1,37 @@
+base_image:
+  image: pytorch/pytorch:2.7.0-cuda12.8-cudnn9-runtime
```
Comment (aghilann) [inline examples/whisper/inference/config.yaml:2]: For any demos we publish, I’d like the user’s training job to run as fast as possible. What do you think about pre-building an image with all the requirements for each cookbook example? Downloading large dependencies like vLLM at runtime slows things down.

File: examples/whisper/inference/model/run_vllm.sh [hunk 2]
```diff
@@ -0,0 +1,4 @@
+export CHECKPOINT_HF=baseten-admin/whisper-larger-v3-turbo-minirun3
```
Comment (aghilann) [inline examples/whisper/inference/model/run_vllm.sh:1]: Should we discuss what sort of lift is needed to get to work to store the weights on B10? If this is the specific start command for whisper, I could make the relevant changes to Django and Truss to support this.

Comment (aghilann) [reply examples/whisper/inference/model/run_vllm.sh:1]: Cut a ticket and assign it to me. I'm not going to block your PR on this since it's a function of whenever I get to it :)

File: examples/whisper/training/config.py [hunk 3]
```diff
@@ -13,12 +13,16 @@
     },
     cache_config=definitions.CacheConfig(
         enabled=True,
-    )
+    ),
+    checkpointing_config=definitions.CheckpointingConfig(
```
Comment (aghilann) [inline examples/whisper/training/config.py:17]: Is there any purpose to this given we don't actually write stuff to the checkpointing dir?

**Comment by aghilann** [comment]
As proof the deployment working, I would talk make an API call to the model itself once it comes up.

File: examples/whisper/inference/config.yaml (hunk 1 above)
Comment (nidhihrmth) [reply examples/whisper/inference/config.yaml]: IMO having a pre-built image for each example has low reward for the effort required. 
Also I'm looking at these examples as just being a starting point for customers using training, and to also drive home the point that users can bring almost any training code to our platform. So IMO having torch or nvidia images makes this clear (without having to reference more files) - I'm down to change this if this isn't the general consensus of course
